src/coreclr/jit/lowerxarch.cpp

```diff
@@ -130,6 +130,36 @@ void Lowering::LowerStoreIndir(GenTreeIndir* node)
             return;
         }
     }
+    else if (!node->HasIndex() && node->gtGetOp2()->OperIs(GT_CNS_DBL))
```

Comment:
question: Why do we need a check `!node->HasIndex()`?

Nit: Because the node is known to be `GT_STOREIND` you can change the signature to `GT_STOREIND`  (or cast it here) and use `node->Data()->OperIs(GT_CNS_DBL)`.